pkg/e2e/openshift/config.go

```diff
@@ -1,139 +1,111 @@
 type E2EConfig struct {
 	// Env defines any environment variable settings in name=value pairs to control the test process
 	Env []string
-
-	// TestCmd determines which suite the runner executes.
-	TestCmd string
-
 	// Tarball determines whether the results should be tar'd or not
 	Tarball bool
-
 	// Suite to be run inside the runner.
 	Suite string
-
 	// TestNames explicitly specify which tests to run as part of the suite. No other tests will be run.
 	TestNames []string
-
 	// Flags to run the suite with.
 	Flags []string
-
 	// Output Dir is where e2e tests serve up results
 	OutputDir string
-
-	Name      string
-	TokenFile string
-	CA        string
+	// Directory with SA creds
+	ServiceAccountDir string
 }
 
 // Cmd returns a shell command which runs the suite.
-func (c E2EConfig) Cmd() string {
-	var cmd bytes.Buffer
-	err := cmdTemplate.Execute(&cmd, c)
-	Expect(err).NotTo(HaveOccurred(), "failed templating command")
-	return cmd.String()
+func (c E2EConfig) GenerateOcpTestCmdBlock() string {
+	cmd := fmt.Sprintf(`
+oc config set-cluster cluster --server=https://kubernetes.default.svc --certificate-authority=%s/ca.crt
+oc config set-credentials user --token=$(cat %s/token)
+oc config set-context cluster --cluster=cluster --user=user
+oc config use-context cluster
+oc config view --raw=true > %s
+export KUBECONFIG=%s
+oc registry login
+ 
+openshift-tests run --dry-run --provider "%s" openshift/conformance/parallel suite  | grep -v "%s" > /tmp/tests
+
+%s openshift-tests run --provider "%s" --file=/tmp/tests %s
+`, c.ServiceAccountDir,
+		c.ServiceAccountDir,
+		getKubeconfigPath(),
+		getKubeconfigPath(),
+		getTestProvider(),
+		getTestSkipRegex(),
+		unwrap(c.Env),
+		getTestProvider(),
+		unwrap(c.Flags))
+	return cmd
 }
 
-func printTests(strs []string) string {
-	testList := strings.Join(strs, "\"\n\"")
-	return fmt.Sprintf("printf '\"%s\"'", testList)
+func getKubeconfigPath() string {
+	return "/tmp/kubeconfig"
 }
 
-func kubeconfigPath() string {
-	if viper.GetString(config.SharedDir) != "" {
-		return viper.GetString(config.SharedDir) + "/kubeconfig"
-	}
-	return ""
+func getTestSkipRegex() string {
+	return viper.GetString(config.Tests.OCPTestSkipRegex)
 }
 
-func region() string {
-	return viper.GetString(config.CloudProvider.Region)
+func unwrap(flags []string) string {
+	return strings.Join(flags, " ")
 }
 
-func cloud() string {
-	if viper.GetString(config.CloudProvider.CloudProviderID) == "gcp" {
-		return "gce"
+// gets zone of cluster. Inferred from node zone of a single zone cluster.
+func getZone() string {
+	zone := ""
+	var k8s *openshift.Client
+	log.SetLogger(ginkgo.GinkgoLogr)
+	var err error
+	k8s, err = openshift.NewFromKubeconfig(viper.GetString(config.Kubeconfig.Path), ginkgo.GinkgoLogr)
+	Expect(err).ShouldNot(HaveOccurred(), "Unable to setup k8s client")
+	kclient, _ := kubernetes.NewForConfig(k8s.GetConfig())
+	nodes, _ := kclient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```

Comment:
```suggestion
	nodes := make(corev1.NodeList)
	_ := k8s.List(context.TODO(), nodes)
```

you can use the k8s client created above